mkversion.sh

```diff
@@ -60,7 +60,7 @@ shift `expr $OPTIND - 1`
 
 version=`git describe --dirty --always --abbrev=7`
 # Add branch name if we only got a hash
-echo "$version" | egrep -q '^[0-9a-f]{7}(-dirty)?$'  &&
+echo "$version" | egrep -q '^.*[0-9a-f]{7}(-dirty)?$'  &&
```

Comment:
This fixed 7 chars is actually wrong, not the `--abbrev` in the previous commit, as the man of `git describe` says:

>       --abbrev=<n>
>           Instead of using the default 7 hexadecimal digits as the abbreviated object name, use <n> digits, or as many digits as needed
>           to form a unique object name. An <n> of 0 will suppress long format, only showing the closest tag.

So it could happen that is actually more. I guess using an explicit `--abbrev` is a good thing too, but this regexp should probably be then `{7,}` (we could make it `{7,40}` but maybe git hashes will grow in the future and I'm not sure if it provides any advantages to have an upper limit).

We could make the explicit `--abbrev` use `4` too, I guess the more concise the version the better.

Reply:
BTW, probably the bug here is to start the regex with `^`, as we only want to match the end of the string. :)

Reply:
Wait! This is supposed to detect when only a hash was given. It shouldn't match a branch name too. If there is a bug, it is elsewhere.

Reply:
Hmm... so do you suggest changing the regex to contain `{7,}` and removing `^.*`? So this line will become:
```
echo "$version" | egrep -q '[0-9a-f]{7,}(-dirty)?$' &&
```
I haven't tested this, but it seems reasonable.

Reply:
I did before I found out that the regex should check for the whole line. Because is actually checking if `git describe` is giving **just a hash**, so it should be kept as `^[0-9a-f]{7,}(-dirty)?$` (the extra `,` is needed because the hash could be actually be longer if 7 chars are not enough to disambiguate the commit).